vector/src/main/java/im/vector/services/EventStreamService.java

```diff
@@ -569,18 +552,7 @@ private void monitorSession(final MXSession session) {
         session.getDataHandler().addListener(new MXEventListener() {
             @Override
             public void onInitialSyncComplete(String toToken) {
-                session.getDataHandler().getStore().post(new Runnable() {
-                    @Override
-                    public void run() {
-                        (new Handler(getMainLooper())).post(new Runnable() {
-                            @Override
-                            public void run() {
-                                refreshForegroundNotification();
-                            }
-                        });
-                    }
-                });
-
+                session.getDataHandler().getStore().post(() -> (new Handler(getMainLooper())).post(() -> refreshForegroundNotification()));
```

Comment:
I'm wondering the reason for these chained `post`...

Reply:
don't know, it's legacy code. If the store thread is not UI it is doing something